drivers/spi/spi_b9x.c

```diff
@@ -160,23 +193,18 @@ static void spi_b91_tx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t l
 			tx = 0;
 		}
 		spi_context_update_tx(ctx, 1, 1);
-		while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_TXF_FULL) {
-		};
-		reg_spi_wr_rd_data(peripheral_id, i % 4) = tx;
+		hal_spi_write(peripheral_id, &tx, 1);
 	}
 }
 
 /* process rx data */
 _attribute_ram_code_sec_
-static void spi_b91_rx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t len)
+static void spi_b9x_rx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t len)
 {
 	uint8_t rx = 0;
 
 	for (int i = 0; i < len; i++) {
-		while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_RXF_EMPTY) {
```

Comment:
why removed?

Reply:
_**hal_spi_read**_(spi_read in hal spi driver) contains the _**while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_RXF_EMPTY)**_ check.